JSDemos/Demos/TreeList/Reordering/description.md

```diff
@@ -1 +1,7 @@
-At&nbsp;first launch, the order of&nbsp;columns in&nbsp;TreeList is&nbsp;the same as&nbsp;the order of&nbsp;objects in&nbsp;the **columns** array. At&nbsp;runtime, a&nbsp;user may be&nbsp;able to&nbsp;reorder columns using drag and drop. To&nbsp;provide the user with this capability, set the **allowColumnReordering** property to _true_.
\ No newline at end of file
+The structure of [dataSource](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#dataSource) objects defines the order of TreeList columns. You can change the order programmatically and allow users to reorder columns in the UI.
+
+### Reorder columns programmatically
+To reorder columns in code, arrange objects and column names as strings in the desired order inside the [columns[]](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/) array. Additionally, you can use the **columns[]**.[visibleIndex](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#visibleIndex) property to specify the position of the column regarding other columns. For details, see the [Column Reordering](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Reordering/) article.
+
+### Allow users to reorder columns in the UI
+To enable column reorder in the UI, set [allowColumnReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#allowColumnReordering) to **true**. If some column should never be reordered, set its [allowReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#allowReordering) property to **false**.
```

Comment:
I added how users reorder columns and change this bit:

> If some column should never be reordered

to make clear what exactly happens when you disable a column's `allowReordering`.

```suggestion
Users can drag and drop column headers to reorder columns. To enable this feature, set the [allowColumnReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#allowColumnReordering) property to **true**. If you do not want users to drag a specific column, disable its [allowReordering](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#allowReordering) property.
```